### PR TITLE
tests: counter_basic_api: Convert to use new dts macros

### PR DESCRIPTION
**Pull request by galak**
Convert DT_RTC_0_NAME to DT_LABEL() references based on the compat being
defined in the build.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: tests/drivers/counter/counter_basic_api/src/test_counter.c [hunk 1]
```diff
@@ -66,8 +66,20 @@ static const char * const devices[] = {
 #if DT_HAS_NODE(DT_INST(0, arm_cmsdk_dtimer))
 	DT_LABEL(DT_INST(0, arm_cmsdk_dtimer)),
 #endif
-#ifdef DT_RTC_0_NAME
-	DT_RTC_0_NAME,
+#if DT_HAS_COMPAT(microchip_xec_timer)
```
Comment (MaureenHelm) [inline tests/drivers/counter/counter_basic_api/src/test_counter.c:69]: why are some lines `DT_HAS_COMPAT` and others `DT_HAS_NODE`?

Comment (galak) [reply tests/drivers/counter/counter_basic_api/src/test_counter.c:69]: The DT_HAS_NODE existed before, I was just reworking the cases for DT_RTC_0_NAME getting set.

I'm sure this is due to whomever updated the test for their given driver support.

Comment (galak) [reply tests/drivers/counter/counter_basic_api/src/test_counter.c]: Rework it so everything but the NRF case is the same now.

File: tests/drivers/counter/counter_basic_api/src/test_counter.c [hunk 2]
```diff
@@ -26,6 +26,13 @@ void *exp_user_data = (void *)199;
 struct counter_alarm_cfg alarm_cfg;
 struct counter_alarm_cfg alarm_cfg2;
 
+#define INST_DT_COMPAT_LABEL(n, compat) DT_LABEL(DT_INST(n, compat)),
+/* Generate a list of LABELs for all instances of the "compat" */
+#define LABELS_FOR_DT_COMPAT(compat) \
```
Comment (erwango) [inline tests/drivers/counter/counter_basic_api/src/test_counter.c:31]: Having an equivalent test dedicated macro (or set of test macros) that could be applied more generally would be nice.

**Review by erwango — APPROVED** [review]

File: tests/drivers/counter/counter_basic_api/src/test_counter.c (hunk 2 above)
Comment (galak) [reply tests/drivers/counter/counter_basic_api/src/test_counter.c:31]: I figured as we see a pattern occur we can refactor this to a common place.

**Review by mbolivar-nordic — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
Tested on mimxrt1050_evk and confirmed the test picks up both GPT instances.

**Pull request merged**